packages/payment-detection/src/any/retrievers/thegraph.ts

```diff
@@ -46,43 +46,49 @@ export class TheGraphAnyToErc20Retriever
    */
   public async getTransferEvents(): Promise<PaymentTypes.ERC20PaymentNetworkEvent[]> {
     const variables = {
-      contractAddress: this.conversionProxyContractAddress,
+      contractAddress: this.contractAddress,
       reference: utils.keccak256(`0x${this.paymentReference}`),
       to: this.toAddress,
-      acceptedTokens: this.acceptedTokens,
       maxRateTimespan: this.maxRateTimespan,
     };
     // Parses, filters and creates the events from the logs with the payment reference
     const events = await this.client.GetConversionPayments(variables);
     // Creates the balance events
-    return events.payments.map((payment) => {
-      const chainlinkDecimal = 8;
-      const decimalPadding = chainlinkDecimal - this.requestCurrency.decimals;
-      const amountWithRightDecimal = BigNumber.from(payment.amount)
-        .div(10 ** decimalPadding)
-        .toString();
-      const feeAmountWithRightDecimal = payment.feeAmount
-        ? BigNumber.from(payment.feeAmount)
-            .div(10 ** decimalPadding)
-            .toString()
-        : undefined;
+    return events.payments
+      .filter((payment) => {
+        if (this.acceptedTokens && this.acceptedTokens.length > 0) {
+          return this.acceptedTokens.includes(payment.tokenAddress?.toLowerCase());
+        }
+        return !payment.tokenAddress;
+      })
+      .map((payment) => {
+        const chainlinkDecimal = 8;
+        const decimalPadding = chainlinkDecimal - this.requestCurrency.decimals;
```

Comment:
Shouldn't use this method instead? https://github.com/RequestNetwork/requestNetwork/blob/3b975067706cd6201fc0cf065b0e33c6bab494ac/packages/payment-detection/src/utils.ts#L29-L35

I think this will break if for whatever reason we have an invoice denominated in ERC20 with conversion.

Reply:
I haven't touched this logic... the diff is misleading :) is it ok to leave it like this? I'm not super aware of chainlink padding. Or, feel free to submit the change yourself

Reply:
Oh yes indeed I got confused by the diff, my bad.